packages/api-reference/src/features/Operation/components/ContentTypeSelect.vue

```diff
@@ -30,101 +35,58 @@ const contentTypes = computed(() => {
 const selectedContentType = ref<ContentType>(
   prop.defaultValue || (contentTypes.value[0] as ContentType),
 )
+
+const selectedOption = computed({
+  get: () =>
+    options.value.find((option) => option.id === selectedContentType.value),
+  set: (option) => {
+    if (option) selectedContentType.value = option.id as ContentType
+  },
+})
+
+const options = computed(() => {
+  return contentTypes.value.map((type) => ({
+    id: type,
+    label: type,
+  }))
+})
+
+// Content type select style variant based on dropdown availability
+const contentTypeSelect = cva({
+  base: 'font-normal text-c-2 bg-b-2 py-0.75 flex cursor-pointer items-center gap-1 rounded-full text-xs',
+  variants: {
+    dropdown: {
+      true: 'border hover:text-c-1 pl-2 pr-1.5',
+      false: 'px-2',
+    },
+  },
+})
 </script>
 <template>
-  <label
-    class="content-type-select"
-    :class="{ 'content-type-no-select': contentTypes.length <= 1 }">
+  <ScalarListbox
+    v-if="prop?.requestBody && contentTypes.length > 1"
+    v-model="selectedOption"
+    :options="options"
+    placement="bottom-end"
+    @update:modelValue="handleSelectContentType"
```

Comment:
seems dangerous to use `v-model` (alias for `@update:modelValue` and `:modelValue`) and `@update:modelValue`

```suggestion
    :modelValue="selectedOption"
    :options="options"
    placement="bottom-end"
    @update:modelValue="handleSelectContentType"
```

Reply:
oops good catch